packages/web/stores/user-settings/language.tsx

```diff
@@ -0,0 +1,55 @@
+import { action, makeObservable, observable, computed } from "mobx";
+import { FunctionComponent } from "react";
+import { IUserSetting } from ".";
+import React from "react";
+import { LanguageSelect, MenuDropdownIconItem } from "../../components/control";
+
+export type LanguageState = { language: string };
+
+const SUPPORTED_LANGUAGES: MenuDropdownIconItem[] = [
+  {
+    value: "en",
+    display: "settings.languages.en",
+    image: "/images/flag-english.png",
+  },
+  {
+    value: "fr",
+    display: "settings.languages.fr",
+    image: "/images/flag-french.png",
+  },
+];
+
+export class LanguageUserSetting implements IUserSetting<LanguageState> {
```

Comment:
Nice! Let me know if you can think of any other improvements that can be made to this user setting abstraction.